docs/setup/k8s/running-in-production-k8s.md

```diff
@@ -62,50 +63,65 @@ Below is a quick example of how to setup the a specific stable version of nuclio
+If you select to handle the implementation yourself, follow these guidelines; the referenced configurations are all [Helm values](/hack/k8s/helm/nuclio/values.yaml):
 
-That being said, here are a few guidelines to get you on your way:
+- Set `*.image.repository` and `*.image.tag` to [freeze a qualified version](#version-freezing), and ensure that the configured images are accessible to the Kubernetes deployment.
+- Set `*.image.pullPolicy` to `Never` or to `IfNotPresent` to ensure that Kubernetes doesn't try to fetch the images from the web.
+- Set `offline` to `true` to put Nuclio in "offline" mode.
+- Set `dashboard.baseImagePullPolicy` to `Never`.
+- Set `registry.pushPullUrl` to a registry URL that's reachable from your system.
+- To use a Nuclio templates library (optional), set `dashboard.templatesArchiveAddress` to a location for serving the templates.
+  Note that you need to package the templates yourself and ensure that they're served locally in a way that's accessible to your system.
 
-- Most definitely [version freeze](#version-freezing). Also, set `*.image.pullPolicy` to `Never` or `IfNotPresent` to make sure k8s won't try to access the web to fetch images at any point in time.
-- Set `Values.offline=true` in the helm values, to put nuclio in "offline" mode. Set `dashboard.baseImagePullPolicy=Never`.
-- Needless to say, in this scenario, you will have to configure nuclio with `registry.pushPullUrl` which is reachable from your system.
-- The processor and onbuild images will also have to be accessible to the dashboard in your environment, as they are required for the building process - (by `docker build`, or [kaniko](#using-kaniko-as-an-image-builder)).
-  This can be tricky as you have to either make those images available to the k8s docker daemon or pull-able from a reachable registry, where they should be preloaded. Use `Values.registy.defaultBaseRegistryURL` to point nuclio at searching in your registry for those images, rather then at the default location of `quay.io/nuclio`.
-  To save some work on setting up a registry and preloading the onbuild images to it (or as a reference to what it should include) - take a look at the [prebaked-registry](https://github.com/nuclio/prebaked-registry).
-- For the Nuclio templates library to be available to you, you'll have to package that yourself, and have it served locally, somewhere within reach of your system. To point Nuclio to it, set `Values.dashboard.templatesArchiveAddress` to where you serve the templates.
+<a id="air-gapped-envir-processor-n-on-build-images"></a>In addition, the processor and on-build images must also be accessible to the dashboard in your environment, as they're required for the build process (either by `docker build` or [Kaniko](#using-kaniko-as-an-image-builder)).
```

Comment:
This needs to follow line 101 (`registry.pushPullUrl`), `templatesArchiveAddress` can come afterwards

Reply:
The reason that I moved it out of the bulleted list was that I tried to keep the list phrasing parallel ("Set xxx ..."), as good writing conventions dictate; that's also why I initially phrased the `dashboard.templatesArchiveAddress` bullet so that "set" comes as soon as possible. But the processor and on-line images bullet is different because it doesn't refer only to values configurations. => Are you adamant that we can't keep the current structure?

Reply:
Yes, since this follows both in content and in importance. It is in close affinity to setting `pushPullUrl` and both have to do with making images available to your cluster (the first referring to where result images will be pushed and this refers to where the source images (onbuild, base) are to be pulled from.
The templatesArchive are a different kind of beast and are also less critical to the core functionality

Reply:
OK, I moved it back to be the bulleted list and slightly rephrased the opening sentence accordingly — see commit ffde717.